psi4/driver/driver_cbs.py

```diff
@@ -916,58 +1077,208 @@ def return_energy_components():
+    elif nbasis == 2 and xtpl_type in ["scf", "corl", "fctl", "dh"]:
+        if xtpl_type == "scf":
+            return xtpl_exponential_2
+        elif xtpl_type == "corl":
+            return xtpl_power_2
+        elif xtpl_type == "fctl":
+            return xtpl_expsqrt_2
+        elif xtpl_type == "dh":
+            return xtpl_power_2
         else:
-            raise ValidationError(f"Wrong number of basis sets supplied to scf_xtpl: {nbasis}")
-    elif xtpl_type == "corl":
-        if nbasis == 2:
-            return corl_xtpl_helgaker_2
+            raise ValidationError(f"Stage treatment must be one of ['scf','corl','fctl','dh'], not '{xtpl_type}'")
+    elif nbasis == 3:
+        if xtpl_type == "scf":
+            return xtpl_exponential_3
+        elif xtpl_type == "fctl":
+            return xtpl_expsqrt_3
         else:
-            raise ValidationError(f"Wrong number of basis sets supplied to corl_xtpl: {nbasis}")
+            raise ValidationError(f"Stage treatment must be one of ['scf','fctl'], not '{xtpl_type}'")
     else:
-        raise ValidationError(f"Stage treatment must be 'corl' or 'scf', not '{xtpl_type}'")
+        raise ValidationError(f"Wrong number of basis sets supplied to scf_xtpl: {nbasis}")
+
+
+def _get_dfa_alpha(xtpl_type: str, bdata: List, funcname: str) -> float:
+    """ A helper function to determine default extrapolation alpha for DFT.
+
+    The parameters for the 'fctl' component are from Kraus, [1]_ based on a fit
+    using numerical results of 30 singlet diatomics with PBE-like functionals
+    with varying amount of HF exchange and MP2 correlation.
+
+    The 'dh' values are based on the standard correlation alpha of 3.0 for [3,4]
+     extrapolation, [2]_ and Truhlar's MP2 alpha of 2.2 for [2,3] extrapolation. [3]_
+     
+    Parameters
+    ----------
+    xtpl_type
+        {'fctl', 'dh'}
+        Extrapolation type: 'fctl' for the functional energy,
+        'dh' for the double-hybrid correlation component.
+    bdata
+        Names and zetas of the basis sets to extrapolate from.
+    funcname
+        Name of the functional.
+
+    Returns
+    -------
+    float
+        Extrapolation alpha to be used.
+
+    References
+    ----------
+    .. [1] Kraus, J. Chem. Theor. Comput. 17 (2021) 5651-5660,
+       DOI: 10.1021/acs.jctc.1c00542
+    .. [2] Halkier, Helgaker, Jorgensen, Klopper, Koch, Olsen, & Wilson,
+       Chem. Phys. Lett. 286 (1998) 243-252,
+       DOI: 10.1016/S0009-2614(99)00179-7
+    .. [3] Truhlar, Chem. Phys. Lett. 294 (1998) 45-48,
+       DOI: 10.1016/S0009-2614(98)00866-5
+    """
+    bnames, bzetas = bdata
+    if funcname not in functionals:
+        raise ValidationError(f"Functional name {funcname} is undefined.")
+    elif xtpl_type == "fctl":
+        if len(set(["cc-pvdz", "cc-pvtz", "cc-pvqz", "cc-pv5z", "cc-pv6z"]).intersection(bnames)) >= 2:
```

Comment:
I won't insist on this change, but in case you don't know this trick, there is an in-line set constructor, `{"cc-pvdz", "cc-pvtz", "cc-pvqz", ...}`. Some don't like it because it reminds them too much of dict comprehensions.

Reply:
Cool trick.